File: src/main/java/Entites/Participation.java
package Entites;

import java.sql.Date;

public class Participation {
    private int id_participation;        // matches DB column: id_participation
    private int id_event;                 // matches DB column: id_event
    private int id_user;                  // matches DB column: id_user
    private Date date_participation;      // matches DB column: date_participation
    private String statut;
    private String nom_participant;       // matches DB column: nom_participant
    private String prenom_participant;    // matches DB column: prenom_participant
    private String email_participant;     // matches DB column: email_participant
    private String telephone;
    private int nombre_places;            // matches DB column: nombre_places

    // For join with Event
    private String eventTitle;

    // Constructors
    public Participation() {}

    public Participation(int id_participation, int id_event, int id_user, Date date_participation,
                         String statut, String nom_participant, String prenom_participant,
                         String email_participant, String telephone, int nombre_places) {
        this.id_participation = id_participation;
        this.id_event = id_event;
        this.id_user = id_user;
        this.date_participation = date_participation;
        this.statut = statut;
        this.nom_participant = nom_participant;
        this.prenom_participant = prenom_participant;
        this.email_participant = email_participant;
        this.telephone = telephone;
        this.nombre_places = nombre_places;
    }

    // ==================== ORIGINAL GETTERS/SETTERS (with underscores) ====================

    public int getId_participation() { return id_participation; }
    public void setId_participation(int id_participation) { this.id_participation = id_participation; }

    public int getId_event() { return id_event; }
    public void setId_event(int id_event) { this.id_event = id_event; }

    public int getId_user() { return id_user; }
    public void setId_user(int id_user) { this.id_user = id_user; }

    public Date getDate_participation() { return date_participation; }
    public void setDate_participation(Date date_participation) { this.date_participation = date_participation; }

    public String getStatut() { return statut; }
    public void setStatut(String statut) { this.statut = statut; }

    public String getNom_participant() { return nom_participant; }
    public void setNom_participant(String nom_participant) { this.nom_participant = nom_participant; }

    public String getPrenom_participant() { return prenom_participant; }
    public void setPrenom_participant(String prenom_participant) { this.prenom_participant = prenom_participant; }

    public String getEmail_participant() { return email_participant; }
    public void setEmail_participant(String email_participant) { this.email_participant = email_participant; }

    public String getTelephone() { return telephone; }
    public void setTelephone(String telephone) { this.telephone = telephone; }

    public int getNombre_places() { return nombre_places; }
    public void setNombre_places(int nombre_places) { this.nombre_places = nombre_places; }

    public String getEventTitle() { return eventTitle; }
    public void setEventTitle(String eventTitle) { this.eventTitle = eventTitle; }

    // ==================== HELPER METHODS for JavaFX PropertyValueFactory (without underscores) ====================

    // These methods allow your FXML to use property names like "idParticipation", "idEvent", etc.

    public int getIdParticipation() { return id_participation; }
    public void setIdParticipation(int idParticipation) { this.id_participation = idParticipation; }

    public int getIdEvent() { return id_event; }
    public void setIdEvent(int idEvent) { this.id_event = idEvent; }

    public int getIdUser() { return id_user; }
    public void setIdUser(int idUser) { this.id_user = idUser; }

    public Date getDateParticipation() { return date_participation; }
    public void setDateParticipation(Date dateParticipation) { this.date_participation = dateParticipation; }

    public String getNomParticipant() { return nom_participant; }
    public void setNomParticipant(String nomParticipant) { this.nom_participant = nomParticipant; }

    public String getPrenomParticipant() { return prenom_participant; }
    public void setPrenomParticipant(String prenomParticipant) { this.prenom_participant = prenomParticipant; }

    public String getEmailParticipant() { return email_participant; }
    public void setEmailParticipant(String emailParticipant) { this.email_participant = emailParticipant; }

    public int getNombrePlaces() { return nombre_places; }
    public void setNombrePlaces(int nombrePlaces) { this.nombre_places = nombrePlaces; }

    @Override
    public String toString() {
        return "Participation{" +
                "id_participation=" + id_participation +
                ", id_event=" + id_event +
                ", eventTitle='" + eventTitle + '\'' +
                '}';
    }
}

File: src/Entity/Participation.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity]
class Participation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: "integer")]
    private ?int $id_participation = null;

    #[ORM\ManyToOne(targetEntity: Event::class, inversedBy: "participations")]
    #[ORM\JoinColumn(name: 'id_event', referencedColumnName: 'id_event', onDelete: 'CASCADE')]
    #[Assert\NotNull(message: "Event is required")]
    private ?Event $id_event = null;

    #[ORM\Column(type: "integer")]
    #[Assert\NotBlank(message: "User ID is required")]
    private int $id_user = 0;

    #[ORM\Column(type: "datetime")]
    #[Assert\NotBlank(message: "Date is required")]
    private ?\DateTimeInterface $date_participation = null;

    #[ORM\Column(type: "string", length: 50)]
    #[Assert\NotBlank(message: "Status is required")]
    private string $statut = 'pending';

    #[ORM\Column(type: "string", length: 100)]
    #[Assert\NotBlank(message: "Last name is required")]
    #[Assert\Length(min: 2, minMessage: "Last name must have at least {{ limit }} characters")]
    private string $nom_participant = '';

    #[ORM\Column(type: "string", length: 100)]
    #[Assert\NotBlank(message: "First name is required")]
    #[Assert\Length(min: 2, minMessage: "First name must have at least {{ limit }} characters")]
    private string $prenom_participant = '';

    #[ORM\Column(type: "string", length: 150)]
    #[Assert\NotBlank(message: "Email is required")]
    #[Assert\Email(message: "The email '{{ value }}' is not valid")]
    private string $email_participant = '';

    #[ORM\Column(type: "string", length: 30)]
    #[Assert\NotBlank(message: "Phone number is required")]
    private string $telephone = '';

    #[ORM\Column(type: "integer")]
    #[Assert\NotBlank(message: "Number of seats is required")]
    #[Assert\Positive(message: "Number of seats must be positive")]
    private int $nombre_places = 0;

    #[ORM\Column(type: "string", length: 255, unique: true, nullable: true)]
    private ?string $token = null;

    #[ORM\Column(type: "boolean", options: ["default" => false])]
    private bool $is_used = false;

    // NEW: price fields
    #[ORM\Column(type: "float", nullable: true)]
    private ?float $unit_price = null;

    #[ORM\Column(type: "float", nullable: true)]
    private ?float $total_price = null;

    // ──────────────────────────────────────────────
    // Getters / Setters (camelCase)
    // ──────────────────────────────────────────────
    public function getIdParticipation(): ?int { return $this->id_participation; }
    public function setIdParticipation(?int $id): self { $this->id_participation = $id; return $this; }

    public function getIdEvent(): ?Event { return $this->id_event; }
    public function setIdEvent(?Event $event): self { $this->id_event = $event; return $this; }

    public function getIdUser(): int { return $this->id_user; }
    public function setIdUser(int $id): self { $this->id_user = $id; return $this; }

    public function getDateParticipation(): ?\DateTimeInterface { return $this->date_participation; }
    public function setDateParticipation(?\DateTimeInterface $date): self { $this->date_participation = $date; return $this; }

    public function getStatut(): string { return $this->statut; }
    public function setStatut(string $statut): self { $this->statut = $statut; return $this; }

    public function getNomParticipant(): string { return $this->nom_participant; }
    public function setNomParticipant(?string $nom): self { $this->nom_participant = $nom ?? ''; return $this; }

    public function getPrenomParticipant(): string { return $this->prenom_participant; }
    public function setPrenomParticipant(?string $prenom): self { $this->prenom_participant = $prenom ?? ''; return $this; }

    public function getEmailParticipant(): string { return $this->email_participant; }
    public function setEmailParticipant(?string $email): self { $this->email_participant = $email ?? ''; return $this; }

    public function getTelephone(): string { return $this->telephone; }
    public function setTelephone(?string $tel): self { $this->telephone = $tel ?? ''; return $this; }

    public function getNombrePlaces(): int { return $this->nombre_places; }
    public function setNombrePlaces(int $nb): self { $this->nombre_places = $nb; return $this; }

    public function getToken(): ?string { return $this->token; }
    public function setToken(?string $token): self { $this->token = $token; return $this; }

    public function getIsUsed(): bool { return $this->is_used; }
    public function setIsUsed(bool $used): self { $this->is_used = $used; return $this; }

    public function getUnitPrice(): ?float { return $this->unit_price; }
    public function setUnitPrice(?float $price): self { $this->unit_price = $price; return $this; }

    public function getTotalPrice(): ?float { return $this->total_price; }
    public function setTotalPrice(?float $price): self { $this->total_price = $price; return $this; }

    // ──────────────────────────────────────────────
    // Aliases underscore (for legacy templates)
    // ──────────────────────────────────────────────
    public function getId_participation(): ?int { return $this->id_participation; }
    public function setId_participation($value): self { $this->id_participation = $value; return $this; }

    public function getId_event(): ?Event { return $this->id_event; }
    public function setId_event($value): self { $this->id_event = $value; return $this; }

    public function getId_user(): int { return $this->id_user; }
    public function setId_user($value): self { $this->id_user = $value; return $this; }

    public function getDate_participation(): ?\DateTimeInterface { return $this->date_participation; }
    public function setDate_participation($value): self { $this->date_participation = $value; return $this; }

    public function getNom_participant(): string { return $this->nom_participant; }
    public function setNom_participant($value): self { $this->nom_participant = $value ?? ''; return $this; }

    public function getPrenom_participant(): string { return $this->prenom_participant; }
    public function setPrenom_participant($value): self { $this->prenom_participant = $value ?? ''; return $this; }

    public function getEmail_participant(): string { return $this->email_participant; }
    public function setEmail_participant($value): self { $this->email_participant = $value ?? ''; return $this; }

    public function getNombre_places(): int { return $this->nombre_places; }
    public function setNombre_places($value): self { $this->nombre_places = $value; return $this; }

    public function getUnit_price(): ?float { return $this->unit_price; }
    public function setUnit_price($value): self { $this->unit_price = $value; return $this; }

    public function getTotal_price(): ?float { return $this->total_price; }
    public function setTotal_price($value): self { $this->total_price = $value; return $this; }
}